target/linux/ath79/dts/qca9558_comfast_wr650ac.dts

```diff
@@ -0,0 +1,147 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+/dts-v1/;
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+#include "qca9557.dtsi"
+
+/ {
+	compatible = "comfast,cf-wr650ac-v1", "qca,qca9557";
+	model = "Comfast CF-WR650AC v1";
+
+	chosen {
+                bootargs = "console=ttyS0,115200n8";
+        };
+
+	leds {
+		compatible = "gpio-leds";
+
+		wps {
+			label = "cf-wr650ac:blue:wps";
+			gpios = <&gpio 20 GPIO_ACTIVE_LOW>;
+		};
+
+		network {
+			label = "cf-wr650ac:blue:network";
+			gpios = <&gpio 4 GPIO_ACTIVE_LOW>;
+		};
+
+		wlan2g {
+			label = "cf-wr650ac:blue:wlan2g";
+			gpios = <&gpio 13 GPIO_ACTIVE_LOW>;
+		};
+
+		wlan5g {
+			label = "comfast:blue:wlan5g";
+			gpios = <&gpio 2 GPIO_ACTIVE_LOW>;
+		};
+	};
+
+	keys {
+		compatible = "gpio-keys-polled";
+		poll-interval = <100>;
+
+		button0 {
+			label = "reset";
+			linux,code = <KEY_RESTART>;
+			gpios = <&gpio 19 GPIO_ACTIVE_LOW>;
+			debounce-interval = <60>;
+		};
+	};
+
+};
+
+&pinmux {
+	pinmux_pa_dcdc_pins {
+		pinctrl-single,bits = <0x0 0xff00 0x0>;
+	};
+
+	pinmux_pa_high_pins {
+		pinctrl-single,bits = <0x10 0xff 0x0>;
+	};
+};
+
+&pcie0 {
+	status = "okay";
+};
+
+&uart {
+	status = "okay";
+};
+
+&spi {
+	status = "okay";
+	num-cs = <1>;
+
+	flash@0 {
+		compatible = "jedec,spi-nor";
+		reg = <0>;
+		spi-max-frequency = <25000000>;
+
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			partition@0 {
+				label = "u-boot";
+				reg = <0x000000 0x020000>;
+				read-only;
+			};
+
+			partition@1 {
+				label = "art";
+				reg = <0x020000 0x010000>;
+				read-only;
+			};
+
+			partition@2 {
+                                label = "firmware";
+                                reg = <0x030000 0xfc0000>;
+                        };
+
+			partition@3 {
+				label = "art-backup";
+				reg = <0xff0000 0x010000>;
+				read-only;
+			}
+		};
+	};
+};
+
+&mdio0 {
+	status = "okay";
+
+	phy0: ethernet-phy@0 {
+		reg = <0>;
+		phy-mode = "rgmii-id";
+	};
+};
+
+&mdio1 {
+	status = "okay";
+
+	phy1: ethernet-phy@1 {
+		reg = <1>;
+		phy-mode = "sgmii";
+	};
+};
+
+&eth0 {
+	status = "okay";
+
+	pll-data = <0xa6000000>;
```

Comment:
@grosjo this is probably wrong, you should provide settings for 1000/100/10 speeds.

Reply:
@ynezz vendor says "it is setup by default at uboot time", so I commented the line

Reply:
This is probably wrong approach, please provide proper `pll-data` values as done in ar71xx. Values from bootloader are going to work until first soft reset, where the PLL register values would be set to defaults.

Reply:
I agree with you. I need to figure out where to get it (vendor not very helpful)